### DISCUSSION TITLE
Container hat Problem mit: Checking Database connection... Failed.

### DISCUSSION THREAD

**Discussion by mibr85** [buanet/ioBroker.docker#402] [Questions & Answers / Fragen & Antworten] [answered]
Hallo zusammen
ich habe versucht einen älteren Container (8.1.0) wieder zu starten, allerdings wird mir ein Problem mit der Datenbank angezeigt.
Kann ich das irgendwie beheben?

 ```
--------------------------------------------------------------------------------

-------------------------     2023-11-12 15:17:36      -------------------------

--------------------------------------------------------------------------------

-----                                                                      -----

----- ██╗  ██████╗  ██████╗  ██████╗   ██████╗  ██╗  ██╗ ███████╗ ██████╗  -----

----- ██║ ██╔═══██╗ ██╔══██╗ ██╔══██╗ ██╔═══██╗ ██║ ██╔╝ ██╔════╝ ██╔══██╗ -----

----- ██║ ██║   ██║ ██████╔╝ ██████╔╝ ██║   ██║ █████╔╝  █████╗   ██████╔╝ -----

----- ██║ ██║   ██║ ██╔══██╗ ██╔══██╗ ██║   ██║ ██╔═██╗  ██╔══╝   ██╔══██╗ -----

----- ██║ ╚██████╔╝ ██████╔╝ ██║  ██║ ╚██████╔╝ ██║  ██╗ ███████╗ ██║  ██║ -----

----- ╚═╝  ╚═════╝  ╚═════╝  ╚═╝  ╚═╝  ╚═════╝  ╚═╝  ╚═╝ ╚══════╝ ╚═╝  ╚═╝ -----

-----                                                                      -----

-----              Welcome to your ioBroker Docker container!              -----

-----                    Startupscript is now running!                     -----

-----                          Please be patient!                          -----

--------------------------------------------------------------------------------

 

--------------------------------------------------------------------------------

-----                          System Information                          -----

-----                    arch:                x86_64                       -----

-----                    hostname:            ioBrokerTWS                  -----

-----                                                                      -----

-----                          Version Information                         -----

-----                    image:               v8.1.0                       -----

-----                    build:               2023-09-21T23:45:32+00:00    -----

-----                    node:                v18.17.1                     -----

-----                    npm:                 9.6.7                        -----

-----                                                                      -----

-----                        Environment Variables                         -----

-----                    SETGID:              1000                         -----

-----                    SETUID:              1000                         -----

--------------------------------------------------------------------------------

 

--------------------------------------------------------------------------------

-----                   Step 1 of 5: Preparing container                   -----

--------------------------------------------------------------------------------

 

This is not the first run of this container. Skipping first run preparation.

 

--------------------------------------------------------------------------------

-----             Step 2 of 5: Detecting ioBroker installation             -----

--------------------------------------------------------------------------------

 

Existing installation of ioBroker detected in "/opt/iobroker".

 

--------------------------------------------------------------------------------

-----             Step 3 of 5: Checking ioBroker installation              -----

--------------------------------------------------------------------------------

 

(Re)setting permissions (This might take a while! Please be patient!)... Done.

 

Fixing "sudo-bug" by replacing sudo with gosu... Done.

 

Checking Database connection... Failed.

Please check your configuration and try again.

For more information see ioBroker Docker Image Docs (https://docs.buanet.de/iobroker-docker-image/docs).

 

This Script will exit now.
```

**Comment by buanet** [comment]
Du kannst versuchen mit der Umgebungsvariable DEBUG = true zu verhindern, dass der Container nach dem Fehler beendet wird. Dann kannst du auf die Kommandozeile des Containers zugreifen und analysieren was das Problem ist.

MfG,
André

**Reply by mibr85** [reply]
Das bekomme ich als Antwort:
```
Checking Database connection... Failed.

[DEBUG] Error message: 
[DEBUG] /opt/scripts/iobroker_startup.sh: /usr/bin/iob: /usr/bin/bash: bad interpreter: No such file or directory
```

Ich verstehe das so, das ihm Dateien Fehlen, aber warum sind die plötzlich aus meinem Volumen verschwunden?

Das ist der Ordnerinhalt:
```
root@ioBrokerTWS:/opt/iobroker# ls -la
total 4404
drwxrwxr-x   8 iobroker iobroker     4096 Nov 14 12:33 .
drwxr-xr-x   1 root     root         4096 Sep 22 01:49 ..
drwxrwxr-x   2 iobroker iobroker     4096 Nov 12 02:40 backups
drwxr-xr-x   4 iobroker iobroker     4096 Sep 24 13:38 .cache
-rw-------   1 iobroker iobroker 12537856 Nov 12 14:43 core
-rwxrwxrwx   1 iobroker iobroker      131 Sep 22 01:49 INSTALLER_INFO.txt
lrwxrwxrwx   1 iobroker iobroker       22 Sep 22 01:49 iob -> /opt/iobroker/iobroker
-rwxr-xr-x   1 iobroker iobroker     2832 Nov 14 12:33 iobroker
-rwxr-xr-x   1 iobroker iobroker     2832 Nov 13 13:37 iobroker.bak
drwxrwxr-x  10 iobroker iobroker     4096 Nov 12 13:59 iobroker-data
drwxrwxr-x   2 iobroker iobroker     4096 Nov 12 14:30 log
drwxrwxr-x 896 iobroker iobroker    36864 Nov 12 14:14 node_modules
drwxr-xr-x   6 iobroker iobroker     4096 Sep 24 13:19 .npm
-rw-rwxr--   1 iobroker iobroker      155 Sep 22 01:48 .npmrc
-rw-rwxr--   1 iobroker iobroker     1103 Nov 12 14:14 package.json
-rw-rwxr--   1 iobroker iobroker   588315 Nov 12 14:14 package-lock.json
root@ioBrokerTWS:/opt/iobroker#
```

**Reply by buanet** [reply]
Ich habe leider keine Idee was da bei dir schief geht. Sieht so aus, als ob das Startscript das Kommando `iob` nicht ausführen kann. Ich habe das eben versucht nachzustellen. Bei mir funktioniert mit dem Image v8.1.0 alles wie es soll...
Einziger Vorschlag den ich noch habe, setze einen frischen ioBroker mit Image v8.1.0 auf und mache einen Restore aus einem Backup. Achte darauf, dass die js-controller Version zum Backup passt oder mache den Restore über die Kommandozeile mit dem Parameter `--force`.

MfG,
André

**Comment by winnyschuster** [comment]
wurde das OS geupdated? Ich denke es liegt an `/usr/bin/bash` welche sich noch(?!!) in `/bin/bash` befindet. Ein Abändern der ersten Zeile in /opt/iobroker/iobroker zu`#!/bin/bash` sollte da helfen, hatte das auch einmal festgestellt

**Reply by buanet** [reply]
Danke für den Hinweis! Das könnte es sein. 

@mibr85 Du hast in deinem ioBroker Verzeichnis eine Datei `iobroker` und eine `iobroker.bak`. Bitte benenne `iobroker` einmal um und entferne `.bak` von `iobroker.bak`. Dann den Container neu starten. 
Dein Problem kommt, weil du v9 getestet hast und jetzt wieder zurück willst...

MfG,
André

**Comment by winnyschuster** [comment]
> Dein Problem kommt, weil du v9 getestet hast und jetzt wieder zurück willst...

exakt, habe das gerade getestet. Die v9 kommt mit OS=bookworm welches die Pfade verändert hat

**Comment by mibr85** [comment]
Leider hat es nicht den gewünschten erfolg gebracht
ich habe mit 
`mv iobroker iobroker.bak1`
dann
`mv iobroker.bak iobroker`
container neustart
container bleibt mit gleicher Fehlermeldung hängen

anschliessend sah es so aus:

root@ioBrokerTWS:/opt/iobroker# ls -la
total 4408
drwxrwxr-x   8 iobroker iobroker     4096 Nov 15 10:49 .
drwxr-xr-x   1 root     root         4096 Sep 22 01:49 ..
drwxrwxr-x   2 iobroker iobroker     4096 Nov 12 02:40 backups
drwxr-xr-x   4 iobroker iobroker     4096 Sep 24 13:38 .cache
-rw-------   1 iobroker iobroker 12537856 Nov 12 14:43 core
-rwxrwxrwx   1 iobroker iobroker      131 Sep 22 01:49 INSTALLER_INFO.txt
lrwxrwxrwx   1 iobroker iobroker       22 Sep 22 01:49 iob -> /opt/iobroker/iobroker
-rwxr-xr-x   1 iobroker iobroker     2832 Nov 15 10:49 iobroker
-rwxr-xr-x   1 iobroker iobroker     2832 Nov 15 10:43 iobroker.bak
-rwxr-xr-x   1 iobroker iobroker     2832 Nov 15 10:44 iobroker.bak1
drwxrwxr-x  10 iobroker iobroker     4096 Nov 12 13:59 iobroker-data
drwxrwxr-x   2 iobroker iobroker     4096 Nov 12 14:30 log
drwxrwxr-x 896 iobroker iobroker    36864 Nov 12 14:14 node_modules
drwxr-xr-x   6 iobroker iobroker     4096 Sep 24 13:19 .npm
-rw-rwxr--   1 iobroker iobroker      155 Sep 22 01:48 .npmrc
-rw-rwxr--   1 iobroker iobroker     1103 Nov 12 14:14 package.json
-rw-rwxr--   1 iobroker iobroker   588315 Nov 12 14:14 package-lock.json
root@ioBrokerTWS:/opt/iobroker#

**Comment by mibr85** [comment]
kann ich irgendwie in die dateien reinschauen? muss ich erst zB nano installieren oder gibt es einen anderen standard editor in dem Container?

**Reply by winnyschuster** [reply]
die Datei `iobroker` ist auch von deinem gemappten Verzeichnis erreichbar. Da wird es dann ein Windowseditor deiner Wahl tun

**Reply by mibr85** [reply, marked as answer]
Hab den Container entsorgt und einen neuen mit einer neuen Dockerinstallation auf mein Synology angelegt und ein Backup eingespielt. jetzt läuft alles
